### PR TITLE
Make message data semi-stringable

### PR DESCRIPTION
**Pull request by trbsi**
Can we make so our Message Data accepts integers not only strings?

### CODE REVIEW

**Comment by jeromegamez** [comment]
Thank you for taking the time to contribute, it's highly appreciated! 🙌 

Unfortunately, this change would not change the messages that are received in the client - Firebase expects the data field names and values to be strings (see https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages#Message.FIELDS.data) - and if they are not strings, Firebase will convert them to strings before delivering them to the clients.

```php
$messaging->send(CloudMessage::fromArray([
	'token' => $token,
	'notification' => [
		'title' => 'Title',
		'body' => 'Body',
	],
	'data' => [
		1 => 1
	]
]));
```

Result:
image

File: src/Firebase/Messaging/MessageData.php [hunk 1]
```diff
@@ -51,9 +51,9 @@ public function jsonSerialize(): array
     /**
      * @param mixed $value
      */
-    private static function isStringable($value): bool
+    private static function isStringableAndNumeric($value): bool
     {
-        return \is_null($value) || \is_scalar($value) || (\is_object($value) && \method_exists($value, '__toString'));
+        return \is_numeric($value) || \is_null($value) || \is_scalar($value) || (\is_object($value) && \method_exists($value, '__toString'));
```
Comment (jeromegamez) [inline src/Firebase/Messaging/MessageData.php:56]: `is_numeric()` is already covered by `is_scalar()`

File: src/Firebase/Messaging/MessageData.php [hunk 2]
```diff
@@ -51,9 +51,9 @@ public function jsonSerialize(): array
     /**
      * @param mixed $value
      */
-    private static function isStringable($value): bool
+    private static function isStringableAndNumeric($value): bool
```
Comment (jeromegamez) [inline src/Firebase/Messaging/MessageData.php:54]: The method would have to check if the value is numeric to make sense: with a value of `"string"`, the method would still return `true`

File: src/Firebase/Messaging/MessageData.php [hunk 3]
```diff
@@ -34,7 +34,7 @@ public static function fromArray(array $data): self
                 );
             }
 
-            $messageData->data[(string) $key] = (string) $value;
+            $messageData->data[$key] = $value;
```
Comment (jeromegamez) [inline src/Firebase/Messaging/MessageData.php:37]: As explained in https://github.com/kreait/firebase-php/pull/488#issuecomment-714423182, the Firebase API does want field names and values to be strings, and will convert non-strings to string when delivering the message to the clients.

Comment (trbsi) [reply src/Firebase/Messaging/MessageData.php:37]: I see, I didn't know that. Thanks for the review

Comment (trbsi) [reply src/Firebase/Messaging/MessageData.php:37]: For that reason I'll close PR

Comment (jeromegamez) [reply src/Firebase/Messaging/MessageData.php:37]: Thank you nonetheless! 🌺

**Pull request closed without merging**